flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/EuclideanDistance.java

```diff
@@ -0,0 +1,272 @@
+package org.apache.flink.ml.classification.knn;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.ml.linalg.DenseVector;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.flink.shaded.curator4.com.google.common.collect.Iterables;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.apache.flink.ml.classification.knn.KnnUtil.appendVectorToMatrix;
+
+/**
+ * Euclidean distance is the "ordinary" straight-line distance between two points in Euclidean
+ * space.
+ *
+ * <p>https://en.wikipedia.org/wiki/Euclidean_distance
+ *
+ * <p>Given two vectors a and b, Euclidean Distance = ||a - b||, where ||*|| means the L2 norm of
+ * the vector.
+ */
+public class EuclideanDistance implements Serializable {
```

Comment:
The design of an independent `EuclideanDistance` is different from both Alink(which implements `FastDistance`) and spark(which does not have such class). Flink ML also has `EuclideanDistanceMeasure` which might help achieving this functionality. It might be better to introduce distance classes while following existing or discussed conventions.

Reply:
you are right, in future we will do this, but not now. because kmeans and knn EuclideanDistance has different feature.
because knn EuclideanDistance do some optimization, just like use matrix accelerate distance calculation.

Reply:
If it's just about adding some optimizations, I believe it is also suitable for kmeans. I still disagree that we should have separate euclidean distance for knn and kmeans.

Reply:
OK, knn's EuclideanDistance may be not a common distance, it's a fast distance. 
Here, I will change the name of knn's distance name, using FastDistance instead of EuclideanDistance.

Reply:
done